### PR TITLE
docs: Add V3 to V4 Parameter Migration Guide

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
# Pull Request Key Information

## Reason for change

Much has changed from LDAR-Sim v3 to LDAR-Sim v4.

## What was changed

Created a new document that outlines the key changes in parameters and inputs from v3 to v4.

## Intended Purpose

Help users navigate the move to V4

## Level of version change required

N/A

## Testing Completed

N/A

## Target Issue

N/A

## Additional Information

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,12 @@ Thomas Barchyn: <tbarchyn@ucalgary.ca>
 
 Chris Hugenholtz: <chhugenh@ucalgary.ca>
 
+## Legacy Versions
+
+LDAR-Sim version 3 can be found on the branch: _TODO_
+
+If moving from V3 to V4, see the [V3 to V4 Parameter Migration Guide](./Version%20Migration%20Guides/V3_to_V4_ParameterMigrationGuide.md) for a useful companion guide to assist in adapting to V4.
```
Comment (SEAJang) [inline README.md:36]: Does this link work?

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
+# LDAR-Sim V4 Parameter Migration Guide
+
+## Purpose
+
+This guide is intended to assist users users who are updating to LDAR-Sim V4 by providing instructions on how to transfer as much of their parameter files as possible to the V4 format from V3. Do note that some new LDAR-Sim behavior has been introduced in V4 and some existing behavior has been changed, thus it is impossible to completely transfer all parameter files and inputs.
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:5]: users users?

Comment (SEAJang) [reply Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:5]: Suggestion: This guide is intended to assist users who wish to update to LDAR-Sim V4 by providing instructions on transferring as much of the existing V3 parameters to V4 format. Do note that new behavior has been introduced in V4 and some existing behavior has been changed, thus, it is impossible to transfer all parameter files and inputs.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+### Virtual World
+
+#### Virtual World: Renamed Parameters
+
+| V3 Parameter | V4 Parameter    | Notes                                                                 |
+|--------------------|-----------|-----------------------------------------------------------------------|
+| infrastructure_file| infrastructure: sites_file  | What was the infrastructure is now just one of the files that define infrastructure. See the [user_manual](../USER_MANUAL.md#infrastructure) on new infrastructure parameters in the virtual world for more information|
+| repair_delay| repairs: delay| repair_delay was moved under the repairs heading and renamed to delay|
+|emissions: LPR| emissions: repairable_emissions: emissions_production_rate| Not quite just a rename, but the functionality is very similar |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:48]: Can we be consistent with the linking - Add links here for the user manual

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+| NRD | emissions: repairable_emissions: duration | Not quite just a rename, but the functionality is very similar |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:49]: Can we be consistent with the linking - Add links here for the user manual

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+| V3 Parameter | Notes                                                                 |
+|--------------------|-----------------------------------------------------------------------|
+| parameter_level | The default value for parameter_level for program parameters has been changed from **program** to **programs**|
+
+#### Programs: Renamed Parameters
+
+| V3 Parameter | V4 Parameter    | Notes                                                                 |
+|--------------------|-----------|-----------------------------------------------------------------------|
+| economics: carbon_price_tonnesCO2e | economics: carbon_price_tonnes_CO2_equivalent |  |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:96]: This parameter was removed - not used in any of the outputs currently.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+The weather file has remained unchanged from V3 to V4.
+
+### Infrastructure
+
+A new, much more granular infrastructure design has been introduced. See [infrastructure](../USER_MANUAL.md#infrastructure) and [virtual world defining files](../USER_MANUAL.md#10-virtual-world-defining-files) sections in the user manual for more information.
+
+### Emissions File (New to V4)
+
+New to V4 is a new file called the Emissions Fie. This file is mandatory to run any V4 simulations. Read more about it in the User Manual [here](../USER_MANUAL.md#emissions-file). It must be set in the virtual world parameters, see the relevant section [here](../USER_MANUAL.md#emissions_file).
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:178]: Linter will likely complain about no new lines at the end of the file.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+No new parameters were introduced in simulation settings.
+
+### Virtual World
+
+#### Virtual World: Renamed Parameters
+
+| V3 Parameter | V4 Parameter    | Notes                                                                 |
+|--------------------|-----------|-----------------------------------------------------------------------|
+| infrastructure_file| infrastructure: sites_file  | What was the infrastructure is now just one of the files that define infrastructure. See the [user_manual](../USER_MANUAL.md#infrastructure) on new infrastructure parameters in the virtual world for more information|
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:46]: Suggestion: The \`infrastructure_file\` is now just one of the files that define the virtual world. See ...

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+|--------------------|-----------|
+| weather_is_hourly| weather functionality has been streamlined  |
+| emissions: consider_venting| emissions granularity has been reworked. See [emissions section](../USER_MANUAL.md#emissions) and all emissions relating parameters in the user manual for more information|
+| emissions: leak_dist_params| Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file)|
+| emissions: leak_dist_type| Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file)|
+|leak_file| Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file)|
+|emissions: leak_file_use|Fit functionality has been removed. This is to avoid scenarios where emissions rates an incorrectly fit to a lognormal distribution when the data is not lognormal|
+| emissions: max_leak_rate | Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file) |
+| emissions: units | Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file) |
+| n_init_leaks_prob | To be reintroduced in later updates as needed |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:63]: Nit: Clearly state removed. And then the rest of the statement.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+| n_init_days | To be reintroduced in later updates as needed |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:64]: Nit: Clearly state removed. And then the rest of the statement.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+|--------------------|-----------|
+| sensor: QE| Replaced with [quantification_error](../USER_MANUAL.md#quantification_error) heading, under which are new quantification error parameters.|
+| mod_loc | Removed as with new object oriented structure type can support this when new sensor types are properly implemented |
+| cost: per_hour | Removed as it was misleading, LDAR-SIm functions on a day timescale |
+| scheduling: LDAR_crew_init_location | Removed all routing parameters as it was not fully functional |
+| scheduling: home_base-files | Removed all routing parameters as it was not fully functional |
+| scheduling: travel_speeds | Removed all routing parameters as it was not fully functional |
+| scheduling: min_time_bt_surveys | New scheduling algorithm uses queue based scheduling, removing the need for this parameter |
+| follow_up: instant_threshold_type | [Instant threshold](../USER_MANUAL.md#instant_threshold) is now fixed to an absolute threshold|
+| follow_up: min_followups| To be reimplemented if needed|
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:150]: Nit: Clearly state removed. And then the rest of the statement.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+| follow_up: min_followup_type| To be reimplemented if needed|
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:151]: Nit: Clearly state removed. And then the rest of the statement.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+| follow_up: min_followup_days_to_end| To be reimplemented if needed|
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:152]: Nit: Clearly state removed. And then the rest of the statement.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+
+#### Methods: Renamed Parameters
+
+| V3 Parameter | V4 Parameter    | Notes                                                                 |
+|--------------------|-----------|-----------------------------------------------------------------------|
+| label| method_name |  |
+| sensor: MDL | sensor: minimum_detection_limit | |
+| n_crews | crew_count| |
+| RS | surveys_per_year | |
+| time | survey_time | |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:132]: t_bt_sites is missing

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+| emissions: max_leak_rate | Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file) |
+| emissions: units | Effectively moved this functionality into the [emissions file](../USER_MANUAL.md#emissions-file) |
+| n_init_leaks_prob | To be reintroduced in later updates as needed |
+| n_init_days | To be reintroduced in later updates as needed |
+
+#### Virtual World: New Parameters
+
+| New Parameter | Notes                                                                   |
+|--------------------|-----------|
+| start_date| moved to virtual world from simulation_settings  |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:70]: Nit: Capitalization

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+| end_date| moved to virtual world from simulation_settings |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:71]: Nit: Capitalization

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+| repairs: cost | moved to virtual world from program parameters |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:72]: Nit: Capitalization

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+| infrastructure_file| infrastructure: sites_file  | What was the infrastructure is now just one of the files that define infrastructure. See the [user_manual](../USER_MANUAL.md#infrastructure) on new infrastructure parameters in the virtual world for more information|
+| repair_delay| repairs: delay| repair_delay was moved under the repairs heading and renamed to delay|
+|emissions: LPR| emissions: repairable_emissions: emissions_production_rate| Not quite just a rename, but the functionality is very similar |
+| NRD | emissions: repairable_emissions: duration | Not quite just a rename, but the functionality is very similar |
+
+#### Virtual World: Removed Parameters
+
+| V3 Parameter | Notes                                                                   |
+|--------------------|-----------|
+| weather_is_hourly| weather functionality has been streamlined  |
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:55]: Nit: Capitalization

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+| emissions: consider_venting| emissions granularity has been reworked. See [emissions section](../USER_MANUAL.md#emissions) and all emissions relating parameters in the user manual for more information|
```
Comment (SEAJang) [inline Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:56]: Nit: Capitalization

**Review by SEAJang — CHANGES_REQUESTED** [review]
Great job overall!

Instead of another folder for Version Migration Guides, could we rename the `Feature Guides` folder to `Other Guides` and move these parameter migration guides in there? 
I feel like there's too many folders... Also, the User manual section needs to be updated - specifically the section called file structure.

File: README.md (hunk 1 above)
Comment (ThomasGalesloot) [reply README.md:36]: Yes it works when I tested it locally

**Comment by ThomasGalesloot** [comment]
> Great job overall!
> 
> Instead of another folder for Version Migration Guides, could we rename the `Feature Guides` folder to `Other Guides` and move these parameter migration guides in there? I feel like there's too many folders... Also, the User manual section needs to be updated - specifically the section called file structure.

Make new folder guides with subfolder for different guide types. I know the user manual is going to be updated at some point, this will need to be update with it.

File: Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md (hunk 6 above)
Comment (ThomasGalesloot) [reply Version Migration Guides/V3_to_V4_ParameterMigrationGuide.md:178]: hmm there is an empty line, might be a git diff view thing?

**Review by SEAJang — APPROVED** [review]

**Pull request merged**